ehrql/backends/tpp.py

```diff
@@ -296,6 +296,32 @@ def modify_query_variables(self, variables):
         """
     )
 
+    covid_therapeutics = QueryTable(
+        """
+        SELECT DISTINCT
+            Patient_ID AS patient_id,
+            COVID_indication AS covid_indication,
+            Count AS count,
+            CurrentStatus AS current_status,
+            Diagnosis AS diagnosis,
+            FormName AS form_name,
+            Intervention AS intervention,
+            CASIM05_date_of_symptom_onset,
+            MOL1_onset_of_symptoms,
+            SOT02_onset_of_symptoms,
+            CASIM05_risk_cohort,
+            MOL1_high_risk_cohort,
+            SOT02_risk_cohorts,
+            STUFF(coalesce(',' + NULLIF(STUFF(coalesce(',' + NULLIF(REPLACE(REPLACE(REPLACE(CASIM05_risk_cohort, 'Patients with a ', ''),  'Patients with ', ''), ' and ', ','), ''), ''), 1, 1, ''), ''), '') + coalesce(',' + NULLIF(STUFF(coalesce(',' + NULLIF(REPLACE(REPLACE(REPLACE(MOL1_high_risk_cohort, 'Patients with a ', ''),  'Patients with ', ''), ' and ', ','), ''), ''), 1, 1, ''), ''), '') + coalesce(',' + NULLIF(STUFF(coalesce(',' + NULLIF(REPLACE(REPLACE(REPLACE(SOT02_risk_cohorts, 'Patients with a ', ''),  'Patients with ', ''), ' and ', ','), ''), ''), 1, 1, ''), ''), ''), 1, 1, '') as risk_cohort,
```

Comment:
urgh this makes my head hurt, and I wrote the previous incantation of it in cohort-extractor
Do you think it could be any clearer by splitting the repeated bit out into a helper function and using it as an f-string value? Or putting some comments in about what it's doing?

Reply:
https://github.com/opensafely-core/ehrql/pull/2025/commits/412b2121df1268cb9d7dce0e1b4226e0a1df1411 better?